Remove backend_debug from torch_core srcs and replace with library dependency

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63111
* #62919

Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

Commit:
Remove backend_debug from torch_core srcs and replace with library dependency



Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

iseeyuan reviewed (approved):
The change in this file should be equivalent to before. The actual usage is in some fb internal code.

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]

Commit:
Update on "Remove backend_debug from torch_core srcs and replace with library dependency"


Differential Revision: [D30259034](https://our.internmc.facebook.com/intern/diff/D30259034/)

### Problem:
Buck contains at least two libraries which have `backend_debug_info.cpp` as a source, `torch_core` and `backend_interface_lib`. `backend_debug_info.cpp` registers BackendDebugInfo as a class. If targets contain both libraries (eg sparkAR debug build with NNAPI delegation), then BackendDebugInfo is registered twice, causing a runtime error.
### Solution:
These changes remove `backend_debug_info.cpp` as a source in torch_core and adds backend_interface_lib as a dependency instead.
On OSS side this looks like a small refactoring of srcs list, so that there's a src list without `backend_debug_info.cpp` and `backend_interface.cpp`, both srcs already included by `backend_interface_lib`.
Please look at internal diff for internal changes as well.

### Test Plan:
Build and run SparkAR internally (`buck build --show-output arstudioplayer_arm64_debug -c pt.enable_nnapi=1` and `buck build --show-output arstudioplayer_arm64_debug`) and internal tests.

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D30259034/)!

[ghstack-poisoned]
